source/server/http_test_server_filter.cc

```diff
@@ -0,0 +1,92 @@
+#include <string>
+
+#include "server/http_test_server_filter.h"
+
+#include "absl/strings/numbers.h"
+
+#include "common/protobuf/utility.h"
+#include "envoy/server/filter_config.h"
+
+#include "api/server/response_options.pb.validate.h"
+
+namespace Nighthawk {
+namespace Server {
+
+HttpTestServerDecoderFilterConfig::HttpTestServerDecoderFilterConfig(
+    nighthawk::server::ResponseOptions proto_config)
+    : server_config_(std::move(proto_config)) {}
+
+HttpTestServerDecoderFilter::HttpTestServerDecoderFilter(
+    HttpTestServerDecoderFilterConfigSharedPtr config)
+    : config_(std::move(config)) {}
+
+void HttpTestServerDecoderFilter::onDestroy() {}
+
+bool HttpTestServerDecoderFilter::mergeJsonConfig(absl::string_view json,
+                                                  nighthawk::server::ResponseOptions& config,
+                                                  std::string& error_message) {
+  error_message = "";
+  try {
+    nighthawk::server::ResponseOptions json_config;
+    Envoy::MessageUtil::loadFromJson(std::string(json), json_config);
+    config.MergeFrom(json_config);
+    Envoy::MessageUtil::validate(config);
+  } catch (Envoy::EnvoyException exception) {
+    error_message = fmt::format("Error merging json config: {}", exception.what());
+  }
+  return error_message.empty();
```

Comment:
Seems possibly fragile, since sometimes exceptions might have an empty string message. Can we make `error_message` an `absl::optional<std::string>`, or something like this?

Reply:
TLDR; done

(Right now, it's not possible to get an emptry string I think, but absl::optional looks like it would be the better api and more resilient towards later breakage when maintaining..)